2.1/Dockerfile

```diff
@@ -1,9 +1,26 @@
 # vim:set ft=dockerfile:
-FROM debian:jessie-backports
+FROM debian:jessie-slim
 
 # explicitly set user/group IDs
 RUN groupadd -r cassandra --gid=999 && useradd -r -g cassandra --uid=999 cassandra
 
+RUN set -ex; \
+	deps=' \
+# solves warning: "jemalloc shared library could not be preloaded to speed up memory allocations"
+		libjemalloc1 \
+# free is used by cassandra-env.sh
+		procps \
+	'; \
+	if ! command -v gpg > /dev/null; then \
+		deps="$deps \
+			dirmngr \
+			gnupg \
+		"; \
+	fi; \
+	apt-get update; \
+	apt-get install -y --no-install-recommends $deps; \
```

Comment:
Do you think this block would be easier to read if it did `apt-get update` at the top and just did two `apt-get install` (the second conditional) instead of using a variable?

Reply:
I was just copying from [php](https://github.com/docker-library/php/blob/be08d2067c6c6b4b9b061a0b082bf8630117a217/Dockerfile-debian.template#L54-L65).

Reply:
But PHP is bad, and it feels bad.

Reply:
```dockerfile
RUN set -eux; \
	apt-get update; \
	apt-get install -y --no-install-recommends \
# solves warning: "jemalloc shared library could not be preloaded to speed up memory allocations"
		libjemalloc1 \
# free is used by cassandra-env.sh
		procps \
	; \
	if ! command -v gpg > /dev/null; then \
￼		apt-get install -y --no-install-recommends \
			dirmngr \
			gnupg \
		; \
	fi; \
	rm -rf /var/lib/apt/lists/*
```